src/GafferScene/PrimitiveSampler.cpp

```diff
@@ -0,0 +1,385 @@
+	const string status = statusPlug()->getValue();
+	if( sourceLocation.empty() || ( primitiveVariables.empty() && status.empty() ) )
+	{
+		return inputObject;
+	}
+
+	PrimitiveVariable::Interpolation outputInterpolation = PrimitiveVariable::Invalid;
+	SamplingFunction samplingFunction = computeSamplingFunction( inputPrimitive, outputInterpolation );
+	if( !samplingFunction || outputInterpolation == PrimitiveVariable::Invalid )
+	{
+		return inputObject;
+	}
+
+	ScenePlug::ScenePath sourcePath;
+	ScenePlug::stringToPath( sourceLocation, sourcePath );
+	if( !SceneAlgo::exists( sourcePlug(), sourcePath ) )
+	{
+		return inputObject;
+	}
+
+	ConstObjectPtr sourceObject = sourcePlug()->object( sourcePath );
+	const Primitive *sourcePrimitive = runTimeCast<const Primitive>( sourceObject.get() );
+	if( !sourcePrimitive )
+	{
+		return inputObject;
+	}
+
+	ConstPrimitivePtr preprocessedSourcePrimitive = sourcePrimitive;
+	if( auto mesh = runTimeCast<const MeshPrimitive>( preprocessedSourcePrimitive.get() ) )
+	{
+		preprocessedSourcePrimitive = MeshAlgo::triangulate( mesh );
+	}
+	PrimitiveEvaluatorPtr evaluator = PrimitiveEvaluator::create( preprocessedSourcePrimitive );
+	if( !evaluator )
+	{
+		return inputObject;
+	}
+
+	PrimitivePtr outputPrimitive = inputPrimitive->copy();
+	const size_t size = outputPrimitive->variableSize( outputInterpolation );
+
+	const string prefix = prefixPlug()->getValue();
+	const M44f transform = inPlug()->fullTransform( path );
+	const M44f sourceTransform = sourcePlug()->fullTransform( sourcePath );
+	const M44f primitiveVariableTransform = sourceTransform * transform.inverse();
+
+	vector<OutputVariableFunction> outputVariables;
+	for( auto &p : preprocessedSourcePrimitive->variables )
+	{
+		if( !StringAlgo::matchMultiple( p.first, primitiveVariables ) )
+		{
+			continue;
+		}
+
+		if( auto o = addPrimitiveVariable( outputPrimitive.get(), prefix + p.first, p.second, outputInterpolation, primitiveVariableTransform ) )
+		{
+			outputVariables.push_back( o );
+		}
+	}
+
+	BoolVectorDataPtr statusData;
+	if( !status.empty() )
+	{
+		statusData = new BoolVectorData();
+		statusData->writable().resize( size, false );
+		outputPrimitive->variables[status] = PrimitiveVariable( outputInterpolation, statusData );
+	}
+
+	PrimitiveEvaluator::ResultPtr evaluatorResult = evaluator->createResult();
+	const M44f samplingTransform = transform * sourceTransform.inverse();
+	for( size_t i = 0; i < size; ++i )
+	{
+		if( samplingFunction( *evaluator, i, samplingTransform, *evaluatorResult ) )
+		{
+			for( const auto &o : outputVariables )
+			{
+				o( i, *evaluatorResult );
+			}
+			if( statusData )
+			{
+				statusData->writable()[i] = true;
+			}
+		}
+	}
+
+	return outputPrimitive;
+}
+
+
+bool PrimitiveSampler::adjustBounds() const
+{
+	return
+		Deformer::adjustBounds() &&
+		prefixPlug()->getValue().empty() &&
+		StringAlgo::matchMultiple( "P", primitiveVariablesPlug()->getValue() )
```

Comment:
Just refreshing my memory, we decided that in the canonical case of `adjustBounds`, we don't consider `width` as requiring adjustment right?

Reply:
Correct.